plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/kafka/ControlInfoAvroSerializer.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.plugins.builtin.kafka
+
+import org.apache.avro.Schema
+import org.apache.avro.generic.{GenericData, GenericRecord}
+import org.apache.commons.io.IOUtils
+import org.slf4j.LoggerFactory
+import za.co.absa.atum.model.{Checkpoint, ControlMeasureMetadata}
+
+import scala.util.control.NonFatal
+
+object ControlInfoAvroSerializer {
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  /**
+   * Converts a control info key represented by a name of a dataset to an Avro record.
+   *
+   * @param datasetName A name of a dataset.
+   * @return An Avro record representing a key to control info topic.
+   */
+  def convertControlInfoKey(datasetName: String): GenericRecord = {
+    val keyAvroSchemaJson = IOUtils.toString(getClass.getResourceAsStream("/info_file_key_avro_schema.avsc"), "UTF-8")
+    val keyAvroSchema = new Schema.Parser().parse(keyAvroSchemaJson)
+    val avroKey = new GenericData.Record(keyAvroSchema)
+
+    avroKey.put("datasetName", datasetName)
+
+    avroKey
+  }
+
+  /**
+   * Converts a control info measurements to an Avro record.
+   *
+   * @param controlInfo An instance of control measurements.
+   * @return An Avro record representing a key to control info topic.
+   */
+  def convertControlInfoRecord(controlInfo: DceControlInfo): Option[GenericRecord] = {
+    val recordAvroSchema = getAvroSchema
+    val avroDceControlInfo = new GenericData.Record(recordAvroSchema)
+
+    try {
+      val avroMetadata = convertMetadata(controlInfo.controlMeasure.metadata, recordAvroSchema)
+      val avroCheckpoints = convertCheckpoints(controlInfo.controlMeasure.checkpoints, recordAvroSchema)
+
+      avroDceControlInfo.put("datasetName", controlInfo.datasetName)
+      avroDceControlInfo.put("datasetVersion", controlInfo.datasetVersion)
+      avroDceControlInfo.put("reportDate", controlInfo.reportDate)
+      avroDceControlInfo.put("reportVersion", controlInfo.reportVersion)
+      avroDceControlInfo.put("runStatus", controlInfo.runStatus)
```

Comment:
These are repeated in `KafkaInfoPlugin` and are prime candidates for constant extraction. (The rest of the String literals in this class might not be repeated elsewhere, but I'd still say they can be extracted as constants to a companion object)

Reply:
Let's discuss this

Reply:
These names come directly from Avro schema. There is a serializer unit test. If either the schema or parameter changes so that they are no longer in sync the unit test will break.

Reply:
That's all right.